### ISSUE TITLE
App sometimes crashes both in ios/android.

### ISSUE THREAD

**Issue by endrits079** [apivideo/api.video-reactnative-live-stream#38]
**Describe the bug**
I have a screen where  I  use <LiveStream/> component like this:

  ```
<LiveStreamView
          isMuted={false}
          video={{
            fps: 30,
            resolution: '720p',
            bitrate: 2 * 1024 * 1024, // # 2 Mbps
          }}
          audio={{
            bitrate: 128000,
            sampleRate: 44100,
            isStereo: true,
          }}
          ref={streamRef}
          style={styles.stream}
          camera={isFront ? 'front' : 'back'}/>
```

This is the startStream function I use inside useEffect to get a key on backend where I stream and then call the startStreaming()

 ```
const startStream = () => {
    liveStreamServices
      .startStream()
      .then(res => res.json())
      .then(data => {
        const stream = data.data;
        streamRef.current?.startStreaming(stream.key, LIVESTREAM_URL);
      });
  };
  
  
    useEffect(() => {
    startStream();
    IdleTimerManager.setIdleTimerDisabled(true);
    return () => onUnmount();
  }, []);
  
```
  
  
Rarely  the app crashes as soon as I get to this screen. I don't know why is it happening and there aren't special steps that I am doing when it is happening so I can't tell how to reproduce this. I am hoping you can  get an understanding from the logs what could be going wrong.

Here are the logs from firebase crashlytics:

Crashed: com.apple.main-thread
0  klipster                       0x123ed8 specialized AVVideoIOUnit.output.getter + 251 (AVVideoIOUnit.swift:251)
1  klipster                       0x118bd4 AVMixer.preferredVideoStabilizationMode.setter + 236 (AVVideoIOUnit.swift:236)
2  klipster                       0x11869c key path setter for AVMixer.preferredVideoStabilizationMode : AVMixer + 4309403292 (<compiler-generated>:4309403292)
3  libswiftCore.dylib             0x1786d0 NonmutatingWritebackBuffer.__deallocating_deinit + 324
4  libswiftCore.dylib             0x3da1d4 _swift_release_dealloc + 56
5  libswiftCore.dylib             0x17a7f8 swift_setAtReferenceWritableKeyPath + 276
6  klipster                       0x111658 specialized Setting.subscript.setter + 41 (Setting.swift:41)
7  klipster                       0x110ad8 specialized Setting.observer.didset + 4309371608
8  klipster                       0x15af5c NetStream.captureSettings.setter + 4309675868 (<compiler-generated>:4309675868)
9  klipster                       0x9160 ApiVideoLiveStream.prepareVideo() + 4308291936 (<compiler-generated>:4308291936)
10 klipster                       0x875c ApiVideoLiveStream.init(initialAudioConfig:initialVideoConfig:preview:) + 145 (ApiVideoLiveStream.swift:145)
11 klipster                       0x402c18 ReactNativeLiveStreamView.audioConfig.didset + 57 (ReactNativeLiveStreamView.swift:57)
12 klipster                       0x402f30 ReactNativeLiveStreamView.audio.didset + 90 (ReactNativeLiveStreamView.swift:90)
13 klipster                       0x402fcc @objc ReactNativeLiveStreamView.audio.setter + 1826316
14 klipster                       0x24d464 __49-[RCTComponentData createPropBlock:isShadowView:]_block_invoke.101 + 315 (RCTComponentData.m:315)
15 klipster                       0x24d618 __49-[RCTComponentData createPropBlock:isShadowView:]_block_invoke_2.102 + 333 (RCTComponentData.m:333)
16 klipster                       0x24d878 __37-[RCTComponentData setProps:forView:]_block_invoke + 371 (RCTComponentData.m:371)
17 CoreFoundation                 0x13948 __NSDICTIONARY_IS_CALLING_OUT_TO_A_BLOCK__ + 24
18 CoreFoundation                 0x9ded0 -[__NSDictionaryM enumerateKeysAndObjectsWithOptions:usingBlock:] + 212
19 klipster                       0x24d7f4 -[RCTComponentData setProps:forView:] + 373 (RCTComponentData.m:373)
20 klipster                       0x2a1fdc __44-[RCTUIManager flushUIBlocksWithCompletion:]_block_invoke + 1199 (RCTUIManager.m:1199)
21 klipster                       0x2a20d4 __44-[RCTUIManager flushUIBlocksWithCompletion:]_block_invoke.426 + 1222 (RCTUIManager.m:1222)
22 libdispatch.dylib              0x24b4 _dispatch_call_block_and_release + 32
23 libdispatch.dylib              0x3fdc _dispatch_client_callout + 20
24 libdispatch.dylib              0x127f4 _dispatch_main_queue_drain + 928
25 libdispatch.dylib              0x12444 _dispatch_main_queue_callback_4CF + 44
26 CoreFoundation                 0x9aa08 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
27 CoreFoundation                 0x7c368 __CFRunLoopRun + 2036
28 CoreFoundation                 0x811e4 CFRunLoopRunSpecific + 612
29 GraphicsServices               0x1368 GSEventRunModal + 164
30 UIKitCore                      0x3a2d88 -[UIApplication _run] + 888
31 UIKitCore                      0x3a29ec UIApplicationMain + 340
32 klipster                       0x75e8 main + 7 (main.m:7)
33 ???                            0x1b3db1948 (Missing)

**Comment by ThibaultBee** [comment]
Hi,

It is a crash in the iOS live stream library. I can't say more from the crash logs.

What version do you use? It seems that this is not the latest version. Could you try with the new version?

Best regards,
Thibault

**Comment by endrits079** [comment]
@ThibaultBee  I upgraded to latest version, but in IOS first time when I want to get to the screen where the LiveStream component is rendered I first ask user for camera and microphone permissions and once permissions accepted then I take to that screen but in xcode I am getting this error only the first time when the user gets to that screen:

2022-19-09 12:17:47.394 [Info] [com.haishinkit.HaishinKit] [AVVideoIOUnit.swift:91] fps > (fps: 30.0, duration: __C.CMTime(value: 100, timescale: 3000, flags: __C.CMTimeFlags(rawValue: 1), epoch: 0))
ApiVideoLiveStream/ApiVideoLiveStream.swift:37: Fatal error: 'try!' expression unexpectedly raised an error: ApiVideoLiveStream.ConfigurationError.invalidParameter("")
2022-09-19 12:17:47.413005+0200 klipster[21613:1973487] ApiVideoLiveStream/ApiVideoLiveStream.swift:37: Fatal error: 'try!' expression unexpectedly raised an error: ApiVideoLiveStream.ConfigurationError.invalidParameter("").

Although in android is not working at all this is the error I am getting:

I/ApiVideoLiveStream: Resolution has been changed from null to 1280x720. Restarting preview.
D/AudioCapture: Not running
D/VideoMediaCodecEncoder: createCodec: {mime=video/avc, width=720, height=1280}
D/AudioCapture: Not running
D/VideoMediaCodecEncoder: createCodec: {mime=video/avc, width=720, height=1280}
E/unknown:ViewManager: Error while updating prop video
    java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:87)
        at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:136)
        at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:56)
        at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:47)
        at com.facebook.react.uimanager.ViewManager.createViewInstance(ViewManager.java:143)
        at com.facebook.react.uimanager.ViewManager.createView(ViewManager.java:76)
        at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:281)
        at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:194)
        at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:909)
        at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1026)
        at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47)
        at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1086)
        at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
        at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175)
        at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1008)
        at android.view.Choreographer.doCallbacks(Choreographer.java:809)
        at android.view.Choreographer.doFrame(Choreographer.java:740)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:995)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:8633)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
     Caused by: java.security.InvalidParameterException: Failed to create unknown codec for: {mime=video/avc, width=720, height=1280}
        at io.github.thibaultbee.streampack.internal.encoders.MediaCodecEncoder.createCodec(MediaCodecEncoder.kt:185)
        at io.github.thibaultbee.streampack.internal.encoders.VideoMediaCodecEncoder.createVideoCodec(VideoMediaCodecEncoder.kt:81)
        at io.github.thibaultbee.streampack.internal.encoders.VideoMediaCodecEncoder.configure(VideoMediaCodecEncoder.kt:59)
        at io.github.thibaultbee.streampack.streamers.bases.BaseStreamer.resetVideo(BaseStreamer.kt:364)
        at io.github.thibaultbee.streampack.streamers.bases.BaseStreamer.stopStream(BaseStreamer.kt:321)
        at io.github.thibaultbee.streampack.streamers.bases.BaseCameraStreamer.stopPreview(BaseCameraStreamer.kt:126)
        at io.github.thibaultbee.streampack.streamers.bases.BaseCameraStreamer.release(BaseCameraStreamer.kt:134)
        at io.github.thibaultbee.streampack.streamers.bases.BaseStreamer.configure(BaseStreamer.kt:243)
        at video.api.livestream.ApiVideoLiveStream.setVideoConfig(ApiVideoLiveStream.kt:86)
        at video.api.reactnative.livestream.ReactNativeLiveStreamView.setVideoConfig(ReactNativeLiveStreamView.kt:57)
        at video.api.reactnative.livestream.ReactNativeLiveStreamViewManager.setVideoConfig(ReactNativeLiveStreamViewManager.kt:62)
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:87) 
        at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:136) 
        at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:56) 
        at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:47) 
        at com.facebook.react.uimanager.ViewManager.createViewInstance(ViewManager.java:143) 
        at com.facebook.react.uimanager.ViewManager.createView(ViewManager.java:76) 
        at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:281) 
        at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:194) 
        at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:909) 
        at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1026) 
        at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47) 
        at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1086) 
        at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29) 
        at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175) 
        at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85) 
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1008) 
        at android.view.Choreographer.doCallbacks(Choreographer.java:809) 
        at android.view.Choreographer.doFrame(Choreographer.java:740) 
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:995) 
        at android.os.Handler.handleCallback(Handler.java:938) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:246) 
        at android.app.ActivityThread.main(ActivityThread.java:8633) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130) 
E/unknown:ReactNative: Exception in native call
    com.facebook.react.bridge.JSApplicationIllegalArgumentException: Error while updating property 'video' of a view managed by: ReactNativeLiveStreamView
        at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:102)
        at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:136)
        at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:56)
        at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:47)
        at com.facebook.react.uimanager.ViewManager.createViewInstance(ViewManager.java:143)
        at com.facebook.react.uimanager.ViewManager.createView(ViewManager.java:76)
        at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:281)
        at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:194)
        at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:909)
        at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1026)
        at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47)
        at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1086)
        at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29)
        at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175)
        at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1008)
        at android.view.Choreographer.doCallbacks(Choreographer.java:809)
        at android.view.Choreographer.doFrame(Choreographer.java:740)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:995)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:246)
        at android.app.ActivityThread.main(ActivityThread.java:8633)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at com.facebook.react.uimanager.ViewManagersPropertyCache$PropSetter.updateViewProp(ViewManagersPropertyCache.java:87)
        at com.facebook.react.uimanager.ViewManagerPropertyUpdater$FallbackViewManagerSetter.setProperty(ViewManagerPropertyUpdater.java:136) 
        at com.facebook.react.uimanager.ViewManagerPropertyUpdater.updateProps(ViewManagerPropertyUpdater.java:56) 
        at com.facebook.react.uimanager.ViewManager.updateProperties(ViewManager.java:47) 
        at com.facebook.react.uimanager.ViewManager.createViewInstance(ViewManager.java:143) 
        at com.facebook.react.uimanager.ViewManager.createView(ViewManager.java:76) 
        at com.facebook.react.uimanager.NativeViewHierarchyManager.createView(NativeViewHierarchyManager.java:281) 
        at com.facebook.react.uimanager.UIViewOperationQueue$CreateViewOperation.execute(UIViewOperationQueue.java:194) 
        at com.facebook.react.uimanager.UIViewOperationQueue$1.run(UIViewOperationQueue.java:909) 
        at com.facebook.react.uimanager.UIViewOperationQueue.flushPendingBatches(UIViewOperationQueue.java:1026) 
        at com.facebook.react.uimanager.UIViewOperationQueue.access$2600(UIViewOperationQueue.java:47) 
        at com.facebook.react.uimanager.UIViewOperationQueue$DispatchUIFrameCallback.doFrameGuarded(UIViewOperationQueue.java:1086) 
        at com.facebook.react.uimanager.GuardedFrameCallback.doFrame(GuardedFrameCallback.java:29) 
        at com.facebook.react.modules.core.ReactChoreographer$ReactChoreographerDispatcher.doFrame(ReactChoreographer.java:175) 
        at com.facebook.react.modules.core.ChoreographerCompat$FrameCallback$1.doFrame(ChoreographerCompat.java:85) 
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1008) 
        at android.view.Choreographer.doCallbacks(Choreographer.java:809) 
        at android.view.Choreographer.doFrame(Choreographer.java:740) 
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:995) 
        at android.os.Handler.handleCallback(Handler.java:938) 
        at android.os.Handler.dispatchMessage(Handler.java:99) 
        at android.os.Looper.loop(Looper.java:246) 
        at android.app.ActivityThread.main(ActivityThread.java:8633) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130) 
     Caused by: java.security.InvalidParameterException: Failed to create unknown codec for: {mime=video/avc, width=720, height=1280}
        at io.github.thibaultbee.streampack.internal.encoders.MediaCodecEncoder.createCodec(MediaCodecEncoder.kt:185)
        at io.github.thibaultbee.streampack.internal.encoders.VideoMediaCodecEncoder.createVideoCodec(VideoMediaCodecEncoder.kt:81)
        at io.github.thibaultbee.streampack.internal.encoders.VideoMediaCodecEncoder.configure(VideoMediaCodecEncoder.kt:59)
        at io.github.thibaultbee.streampack.streamers.bases.BaseStreamer.resetVideo(BaseStreamer.kt:364)
        at io.github.thibaultbee.streampack.streamers.bases.BaseStreamer.stopStream(BaseStreamer.kt:321)
        at io.github.thibaultbee.streampack.streamers.bases.BaseCameraStreamer.stopPreview(BaseCameraStreamer.kt:126)
        at io.github.thibaultbee.streampack.streamers.bases.BaseCameraStreamer.release(BaseCameraStreamer.kt:134)
        at io.github.thibaultbee.streampack.streamers.bases.BaseStreamer.configure(BaseStreamer.kt:243)
        at video.api.livestream.ApiVideoLiveStream.setVideoConfig(ApiVideoLiveStream.kt:86)
        at video.api.reactnative.livestream.ReactNativeLiveStreamView.setVideoConfig(ReactNativeLiveStreamView.kt:57)
        at video.api.reactnative.livestream.ReactNativeLiveStreamViewManager.setVideoConfig(ReactNativeLiveStreamViewManager.kt:62)
E/unknown:ReactNative: 	... 27 more

**Comment by endrits079** [comment]
I was playing with the config values and for the video props I have set resolution to 720p changing this to 360 the app stops crashing in android but the quality is unacceptable  why is crashing with 720p?

**Comment by ThibaultBee** [comment]
For iOS, according to https://github.com/apivideo/api.video-ios-live-stream/blob/e336109873807d37f85962a8658cd5b33de0f246/Sources/Configuration.swift#L53, the resolution is not supported. My guess is somehow HaishinKit default resolution is not supported by these `if`. Could you add a breakpoint or a message in this function to check which resolution is not supported? So I will add it.

For Android, It means that your encoder config is not valid. There might be multiple encoder on your device and somehow the one selected is not the good one. I need to have a deeper look here.

**Comment by ThibaultBee** [comment]
For Android,  the issue is around https://github.com/ThibaultBee/StreamPack/blob/dc2680207dade2ea27c22e77beffb7d9450e00de/core/src/main/java/io/github/thibaultbee/streampack/internal/encoders/MediaCodecEncoder.kt#L183
I don't understand why it does not find a proper AVC encoder...

Could you:
1/ Reproduced the issue with https://github.com/ThibaultBee/StreamPack/suites/7900508529/artifacts/335986000
2/ Then, try to reproduce with https://github.com/ThibaultBee/StreamPack/suites/8371944623/artifacts/369328327

**Comment by Pitros** [comment]
Same happens for me, randomly on starting/stopping streaming or just component mount/unmount.
In basic app sample https://github.com/Pitros/stream-sample
App crashes on Samsung S9 with Android 10, but works flawlessly on Samsung S20 with Android 12.

Here's log from two crashes - first just remounting component, second when trying to start/stop stream.
https://gist.github.com/Pitros/cd4d9e1b5f6fb12b71ba98cf1e1703e0

Let me know if I can provide some better details for you.

Also in emulator app dies instantly with `Unable to instantiate codec 'video/avc'`.

**Comment by ThibaultBee** [comment]
@Pitros 

I don't see the crash on remounting component. I can see a `java.lang.UnsupportedOperationException: surface is not valid` but this exception is catched, so it should not cause any crash.

There is crash on stopStreaming due to this [`Log.w`](https://github.com/apivideo/api.video-reactnative-live-stream/blob/747db656125754147202293c464e481e4052f2da/android/src/main/java/video/api/reactnative/livestream/ReactNativeLiveStreamView.kt#L159.). But I don't see any reason for it to crash like this.

For the emulator, maybe your configuration is not supported by the emulator?

**Comment by rahul2992** [comment]
I can confirm the app crashes randomly on start and stop stream on both iOS and android. 

I have found the following errors on iOS. 
1. When the user gives permissions for the first time and LiveStream is mounted:

`HaishinKit/AVVideoIOUnit.swift:251: Fatal error: Unexpectedly found nil while unwrapping an Optional value
2022-09-24 16:22:12.971988+0530 Socialboat[3629:14580094] HaishinKit/AVVideoIOUnit.swift:251: Fatal error: Unexpectedly found nil while unwrapping an Optional value`

Live stream View. The following component is mounted using an ternary operator. 
```
<LiveStreamView
    ref={cameraRef}
    camera="front"
    video={{
      resolution: "240p",
      bitrate: 800 * 1000, 
    }}
    audio={{
      bitrate: 128 * 1000, 
      sampleRate: 44100,
      isStereo: false,
    }}
    enablePinchedZoom={false}
    onConnectionSuccess={() => {
      console.log("Connection success");
    }}
    onConnectionFailed={(reason: string) => {
      console.log("Connection failed", reason);
    }}
    onDisconnect={() => {
      console.log("disconnected");
    }}
    isMuted={true}
    style={{
      flex: 1,
      zIndex: 10,
      borderRadius: 20,
      backgroundColor: "black",
    }}
  />
```

Also from our live iOS build, I am getting the following crash in Crashlytics. 
```
Fatal Exception: NSGenericException
0  CoreFoundation                 0x99288 __exceptionPreprocess
1  libobjc.A.dylib                0x16744 objc_exception_throw
2  AVFCapture                     0x35a1c -[AVCaptureSession startRunning]
3  Socialboat                     0xbbea6c closure #1 in AVMixer.startRunning() + 293 (AVMixer.swift:293)
4  Socialboat                     0x3537c thunk for @escaping @callee_guaranteed () -> () (<compiler-generated>)
5  libdispatch.dylib              0x1e6c _dispatch_call_block_and_release
6  libdispatch.dylib              0x3a30 _dispatch_client_callout
7  libdispatch.dylib              0x15280 _dispatch_root_queue_drain
8  libdispatch.dylib              0x1596c _dispatch_worker_thread2
9  libsystem_pthread.dylib        0x1080 _pthread_wqthread
10 libsystem_pthread.dylib        0xe5c start_wqthread
```

I am investigating android crashes and will get back soon.

**Comment by ThibaultBee** [comment]
@rahul2992 
I try to reproduce your issue 1. (when user gives permissions for the first time) but I can not reproduced it. Can you reproduced it with the example provided in this code base? Is there a way I can reproduced it easily?

About the second one, I tried to synchronize video config and startStreaming. It might be enough to fix this issue.

**Comment by rahul2992** [comment]
Thank you @ThibaultBee for looking at it. I will try to reproduce the issue in the coming week. As of now we have moved to another solution for handling live stream.

**Comment by tarun-showday** [comment]
I am also facing the similar, It's not happening every time.  

This is the logs for the same. If this logs can help @ThibaultBee to debug the issue: 
https://gist.github.com/tarun-showday/30ad223057e927da3f5c085404b1fae2

**Comment by TheFe91** [comment]
I'm experiencing the crash issue too, but only on Android and only in stop phase. All the rest works fine

**Comment by ThibaultBee** [comment]
@TheFe91 
There are 2 opened issues concerning crashes on `stopStreaming` on Android. Please, comment in the correct issue.

**Comment by ThibaultBee** [comment]
Just released a new version (v1.2.1) that will fix more of the crashes described here.
Please, open new issue if you face a crash with the rule one crash = one issue.
